filebeat/input/syslog/config.go

```diff
@@ -64,7 +71,17 @@ func factory(
 		if err := cfg.Unpack(&config); err != nil {
 			return nil, err
 		}
-		return tcp.New(&config, cb)
+
+		dConfig := defaultDelimiter
+		if err := cfg.Unpack(&dConfig); err != nil {
+			return nil, err
+		}
+
+		splitFunc := tcp.SplitFunc([]byte(dConfig.LineDelimiter))
+		if splitFunc != nil {
+			return nil, fmt.Errorf("error creating splitFunc from delimiter %s", dConfig.LineDelimiter)
+		}
+		return tcp.New(&config, splitFunc, cb)
```

Comment:
@vjsamuel 

The following will work and will be backward compatible with the existing configuration.
The struct structure can be different from the YML structure when you are using ucfg 'inline' option, also I've removed the check for the `tcp.SplitFunc` because we have a **nonzero** validation the factory cannot return NIL.

```diff
diff --git a/filebeat/input/syslog/config.go b/filebeat/input/syslog/config.go
index 4b90223f4..dd9f99b88 100644
--- a/filebeat/input/syslog/config.go
+++ b/filebeat/input/syslog/config.go
@@ -41,9 +41,17 @@ var defaultConfig = config{
 	},
 }
 
-var defaultTCP = tcp.Config{
-	Timeout:        time.Minute * 5,
-	MaxMessageSize: 20 * humanize.MiByte,
+type syslogTCP struct {
+	tcp.Config    `config:",inline"`
+	LineDelimiter string `config:"line_delimiter" validate:"nonzero"`
+}
+
+var defaultTCP = syslogTCP{
+	Config: tcp.Config{
+		Timeout:        time.Minute * 5,
+		MaxMessageSize: 20 * humanize.MiByte,
+	},
+	LineDelimiter: "\n",
 }
 
 var defaultUDP = udp.Config{
@@ -51,14 +59,6 @@ var defaultUDP = udp.Config{
 	Timeout:        time.Minute * 5,
 }
 
-type delimiterConf struct {
-	LineDelimiter string `config:"line_delimiter" validate:"nonzero"`
-}
-
-var defaultDelimiter = delimiterConf{
-	LineDelimiter: "\n",
-}
-
 func factory(
 	cb inputsource.NetworkFunc,
 	config common.ConfigNamespace,
@@ -72,16 +72,9 @@ func factory(
 			return nil, err
 		}
 
-		dConfig := defaultDelimiter
-		if err := cfg.Unpack(&dConfig); err != nil {
-			return nil, err
-		}
+		splitFunc := tcp.SplitFunc([]byte(config.LineDelimiter))
 
-		splitFunc := tcp.SplitFunc([]byte(dConfig.LineDelimiter))
-		if splitFunc != nil {
-			return nil, fmt.Errorf("error creating splitFunc from delimiter %s", dConfig.LineDelimiter)
-		}
-		return tcp.New(&config, splitFunc, cb)
+		return tcp.New(&config.Config, splitFunc, cb)
 	case udp.Name:
 		config := defaultUDP
 		if err := cfg.Unpack(&config); err != nil {
diff --git a/filebeat/input/tcp/input.go b/filebeat/input/tcp/input.go
index 1688af3c4..38eafb4ce 100644
--- a/filebeat/input/tcp/input.go
+++ b/filebeat/input/tcp/input.go
@@ -18,7 +18,6 @@
 package tcp
 
 import (
-	"fmt"
 	"sync"
 	"time"
 
@@ -78,9 +77,6 @@ func NewInput(
 	}
 
 	splitFunc := tcp.SplitFunc([]byte(config.LineDelimiter))
-	if splitFunc == nil {
-		return nil, fmt.Errorf("unable to create splitFunc for delimiter %s", config.LineDelimiter)
-	}
 
 	server, err := tcp.New(&config.Config, splitFunc, cb)
 	if err != nil {

```